refinery/ui/source/js/data-set-import/controllers/isa-tab-import.js

```diff
@@ -28,6 +50,16 @@
     this.closeError = function () {
       this.isErrored = false;
     };
+
+    // helper watcher which updates the upload file status from component
+    $scope.$watchCollection(
+      function () {
+        return fileUploadStatusService.fileUploadStatus;
+      },
```

Comment:
I'm not seeing why the `Collection` part is necessary here: I can see you've moved the real status down into a property, but could you just add it here, too, and use plain `watch`? Sorry not to get it.

Reply:
data-file-edit-modal-ctrl.js for comparison.

Reply:
@mccalluc For this instance, watch would work vs watchCollection since we're not adding any additional properties. watchCollection just tends to be my default for objects. Will adjust.

Reply:
@mccalluc We've discussed this in a previous pull request. Issues with setting primitives in services. https://github.com/refinery-platform/refinery-platform/pull/2921 

Bottom line it a javascript thing when it comes to primitives versus watching references (obj, arrays). Did you get a chance to look at the references I posted previously in pull request 2921?